service/src/test/java/bio/terra/workspace/pact/TpsApiTest.java

```diff
@@ -0,0 +1,555 @@
+            .object("newAttributes", tpsPolicyInputsObjectShape)
+            .stringMatcher("updateMode", updateModeRegex);
+    var updateResponseShape =
+        new PactDslJsonBody()
+            .booleanType("updateApplied")
+            .object("conflicts", new PactDslJsonArray())
+            .object("resultingPao"); // TpsPaoGetResult
+    return builder
+        .given(existingPolicyState)
+        .uponReceiving("A request to update a policy")
+        .method("PUT")
+        .body(updateRequestShape)
+        .headers(contentTypeJsonHeader)
+        .pathFromProviderState(
+            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyProviderStateValue),
+            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyId))
+        .willRespondWith()
+        .status(HttpStatus.OK.value())
+        .body(updateResponseShape)
+        .toPact();
+  }
+
+  @Test
+  @PactTestFor(pactMethod = "replacePaoThatExists")
+  public void replacingAnExistingPaoWithNoConflicts() throws Exception {
+    var newInputs =
+        new TpsPolicyInputs()
+            .inputs(List.of(new TpsPolicyInput().name("new_name").namespace("new_namespace")));
+    var result = dispatch.replacePao(existingPolicyId, newInputs, TpsUpdateMode.FAIL_ON_CONFLICT);
+    assertNotNull(result);
+  }
+
+  @Pact(consumer = "wsm", provider = "tps")
+  public RequestResponsePact updatePaoPreexistingNoConflicts(PactDslWithProvider builder) {
+    var updateRequestShape =
+        new PactDslJsonBody()
+            .object("addAttributes", tpsPolicyInputsObjectShape)
+            .object("removeAttributes", tpsPolicyInputsObjectShape)
+            .stringMatcher("updateMode", updateModeRegex);
+    var updateResponseShape =
+        new PactDslJsonBody()
+            .booleanType("updateApplied")
+            .object("conflicts", new PactDslJsonArray())
+            .object("resultingPao"); // TpsPaoGetResult
+    return builder
+        .given(existingPolicyState)
+        .uponReceiving("A request to update a policy")
+        .method("PATCH")
+        .body(updateRequestShape)
+        .headers(contentTypeJsonHeader)
+        .pathFromProviderState(
+            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyProviderStateValue),
+            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyId))
+        .willRespondWith()
+        .status(HttpStatus.OK.value())
+        .body(updateResponseShape)
+        .toPact();
+  }
+
+  @Test
+  @PactTestFor(pactMethod = "updatePaoPreexistingNoConflicts")
+  public void updatingAnExistingPaoWithNoConflicts() throws Exception {
+    var remove =
+        new TpsPolicyInputs()
+            .inputs(List.of(new TpsPolicyInput().name("test_name").namespace("test_namespace")));
+    var add =
+        new TpsPolicyInputs()
+            .inputs(List.of(new TpsPolicyInput().name("new_name").namespace("new_namespace")));
+    var result = dispatch.updatePao(existingPolicyId, add, remove, TpsUpdateMode.FAIL_ON_CONFLICT);
```

Comment:
We only use the modes `FAIL_ON_CONFLICT' and 'DRY_RUN', and there isn't really any difference in request/response shape between them.
Even though we're only calling it with `FAIL_ON_CONFLICT` in our actual tests, the pact still includes all the values.